homeassistant/components/plaato/sensor.py

```diff
@@ -31,134 +23,62 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     """Set up the Plaato sensor."""
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(hass, entry, async_add_entities):
     """Set up Plaato from a config entry."""
     devices = {}
 
-    def get_device(device_id):
-        """Get a device."""
-        return hass.data[PLAATO_DOMAIN].get(device_id, False)
-
-    def get_device_sensors(device_id):
-        """Get device sensors."""
-        return hass.data[PLAATO_DOMAIN].get(device_id).get(PLAATO_DEVICE_SENSORS)
-
-    async def _update_sensor(device_id):
+    async def _update_sensor(device_id, sensor_data: PlaatoDevice):
         """Update/Create the sensors."""
-        if device_id not in devices and get_device(device_id):
-            entities = []
-            sensors = get_device_sensors(device_id)
-
-            for sensor_type in sensors:
-                entities.append(PlaatoSensor(device_id, sensor_type))
-
-            devices[device_id] = entities
-
-            async_add_entities(entities, True)
+        entry_data = hass.data[DOMAIN][entry.entry_id]
```

Comment:
Move this to the top of function so it can be used every place `hass.data[DOMAIN][entry.entry_id]` is used.